session1/3-challenge/5_string.rb

```diff
@@ -5,4 +5,5 @@
 # add_more_ruby("Sad times are ahead!")         # => "Happy times are ahead!"
 
 def add_more_ruby(string)
+	string.gsub("sad", "happy").gsub("Sad", "Happy")
```

Comment:
I was thinking about this and it seems to me that you are essentially checking the string twice - first for lowercase and then upper case. Why not consider using a regex to locate all occurrences of "sad" and then replace as requried?

Reply:
Have you got an example of what you mean? I would imagine even using a regex would need more logic to check what version of sad/Sad it is and then replacing it with the correct version of happy/Happy. This is maybe how the gsub method is doing it in its implementation.

Reply:
You could pass regex that is non case sensetive like so /sad/i, and then add a hash for replacing sad and Sad. so it would look like this:
string.gsub(/sad/i, 'sad' => 'happy', 'Sad' => 'Happy')

Reply:
Yeah that works. Nice solution! I have to admit my regex wasn't so strong when I was doing this. I probably need to go back through these solutions and look for areas for improvement.